sdk/test/AzureTest/AzureTestTests/AzureTestTests.swift

```diff
@@ -0,0 +1,169 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import XCTest
+import Foundation
+import DVR
+import AzureTest
+
+class AzureTestTests: XCTestCase {
+
+    var fakeData: Data!
+    
+    var fakeRequest: URLRequest!
+    
+    var fakeResponse: URLResponse!
+    
+    var fakeResponseData: Data!
+    
+    private func chooseRecordedInteraction(number: Int) {
+        fakeRequest = fakeData.request(number: number)
+        fakeResponse = fakeData.response(number: number)
+        fakeResponseData = fakeData.responseData(number: number)
+    }
+    
+    override func setUpWithError() throws {
+        let testBundle = Bundle(for: type(of: self))
+        let path = testBundle.path(forResource: "TestData", ofType: "json")
+        fakeData = try! Data(contentsOf: URL(fileURLWithPath: path!))
+        chooseRecordedInteraction(number: 8)
+        
+    }
+
+    func test_scrubbingRequest_removeSubscriptionIDs() throws {
+        let cleanedRequest = Filter().scrubSubscriptionIDs(from: fakeRequest!)
+        let shouldPass = cleanedRequest.url?.absoluteString.contains(regex: Filter.subcriptionIDReplacement) ?? false
+        XCTAssert(shouldPass)
+        
+    }
+
+    func test_scrubbingResponse_removeSubscriptionIDs() throws {
+        let dirtyHeaders = ["location": "[\"https://management.azure.com/subscriptions/72f988bf-86f1-41af-91ab-2d7cd011db47/providers/Microsoft.KeyVault/locations/eastus/operationResults/VVR8MDYzNzU0NDA3MTY0MzE2NTczMnwwNjZENTEwRTA4N0U0MTY5ODc1MDhDRDY3QUJDMzdGOQ?api-version=2019-09-01\"]"]
+        
+        let dirtyBody = """
+            "string": "{\"id\":\"/subscriptions/72f988bf-86f1-41af-91ab-2d7cd011db47/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/myValtZikfikxz\",\"name\":\"myValtZikfikxz\",\"type\":\"Microsoft.KeyVault/deletedVaults\",\"properties\":{\"vaultId\":\"/subscriptions/72f988bf-86f1-41af-91ab-2d7cd011db47/resourceGroups/rgname/providers/Microsoft.KeyVault/vaults/myValtZikfikxz\",\"location\":\"eastus\",\"tags\":{},\"deletionDate\":\"2021-04-19T05:32:42Z\",\"scheduledPurgeDate\":\"2021-07-18T05:32:42Z\"}}"
+        """
+        
+        var shouldPass = false
+        
+        let cleanLocation = Filter.scrubSubscriptionId(from: dirtyHeaders["location"])
+        let cleanBody = Filter.scrubSubscriptionId(from: dirtyBody)
+        
+        shouldPass = (cleanLocation?.contains(regex: Filter.subcriptionIDReplacement) ?? false) && (cleanBody?.contains(regex: Filter.subcriptionIDReplacement) ?? false)
```

Comment:
```suggestion
        let shouldPass = cleanLocation!.contains(regex: Filter.subcriptionIDReplacement) && cleanBody!.contains(regex: Filter.subcriptionIDReplacement)
```

However, as I look at this, perhaps the criteria should actually be that we _don't_ find the original value rather than the fact that we _do_ find the replacement.